### PR TITLE
Get support for squiggles and LB in "AnyCode" workspace

### PR DESCRIPTION
**Pull request by huizhonglong**
1) Add InternalsVisibleTo for "AnyCode" in order to get access to ISolutionCrawlerRegistrationService
2) Add AnyCode workspace kind as one of the kinds that the diagnostic service supports.
@heejaechang @jasonmalinowski

### CODE REVIEW

**Comment by jasonmalinowski** [comment]
Review: @dotnet/roslyn-ide

File: src/Workspaces/Core/Portable/PublicAPI.Unshipped.txt [hunk 1]
```diff
@@ -27,6 +27,7 @@ Microsoft.CodeAnalysis.Editing.OperatorKind.UnaryNegation = 24 -> Microsoft.Code
 Microsoft.CodeAnalysis.Editing.OperatorKind.UnaryPlus = 25 -> Microsoft.CodeAnalysis.Editing.OperatorKind
 Microsoft.CodeAnalysis.Editing.SyntaxGenerator.OperatorDeclaration(Microsoft.CodeAnalysis.IMethodSymbol method, System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.SyntaxNode> statements = null) -> Microsoft.CodeAnalysis.SyntaxNode
 abstract Microsoft.CodeAnalysis.Editing.SyntaxGenerator.OperatorDeclaration(Microsoft.CodeAnalysis.Editing.OperatorKind kind, System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.SyntaxNode> parameters = null, Microsoft.CodeAnalysis.SyntaxNode returnType = null, Microsoft.CodeAnalysis.Accessibility accessibility = Microsoft.CodeAnalysis.Accessibility.NotApplicable, Microsoft.CodeAnalysis.Editing.DeclarationModifiers modifiers = default(Microsoft.CodeAnalysis.Editing.DeclarationModifiers), System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.SyntaxNode> statements = null) -> Microsoft.CodeAnalysis.SyntaxNode
+const Microsoft.CodeAnalysis.WorkspaceKind.AnyCodeRoslynWorkspace = "AnyCodeRoslynWorkspace" -> string
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/PublicAPI.Unshipped.txt]: This should not be exposed publically from our API.

File: src/Workspaces/Core/Portable/Workspace/WorkspaceKind.cs [hunk 2]
```diff
@@ -15,5 +15,6 @@ public static class WorkspaceKind
         public const string MetadataAsSource = nameof(MetadataAsSource);
         public const string MiscellaneousFiles = nameof(MiscellaneousFiles);
         public const string Preview = nameof(Preview);
+        public const string AnyCodeRoslynWorkspace = nameof(AnyCodeRoslynWorkspace);
```
Comment (heejaechang) [inline src/Workspaces/Core/Portable/Workspace/WorkspaceKind.cs]: this is just text, have this in both anycode and roslyn.

File: src/Workspaces/Core/Portable/PublicAPI.Unshipped.txt (hunk 1 above)
Comment (huizhonglong) [inline src/Workspaces/Core/Portable/PublicAPI.Unshipped.txt]: Thanks. I just reverted it.

**Comment by huizhonglong** [comment]
OK I've reverted the InternalsVisibleTo changes. We need the change on DiagnosticAnalyzerService to enable Squiggles in AnyCode for Dev15 **Preview** as clarified on emails. Pls take a look. Thxs!
@heejaechang @jasonmalinowski

**Comment by davkean** [comment]
@dotnet-bot test eta please

**Comment by heejaechang** [comment]
:+1: I will merge this in once eta pass.

**Pull request merged**